### ISSUE TITLE
code about testing

### ISSUE THREAD

**Issue by cmhungsteve** [SSARCandy/DeepCORAL#11]
Hello, 
I have some questions about your testing codes in `main.py`: 
1. 
>     # sum up batch loss
>     test_loss += torch.nn.functional.cross_entropy(out, target, size_average=False).data[0]
>     # get the index of the max log-probability
>     pred = out.data.max(1, keepdim=True)[1]
>     correct += pred.eq(target.data.view_as(pred)).cpu().sum()
 
why don't you directly calculate the loss as you do for training 
like this: `classification_loss = torch.nn.functional.cross_entropy(out1, source_label)`
Instead, you sum up the loss then divide by the times you sum up.
Is there any reason for that?

2.
>     out1, out2 = model(data, data)
>     out = out1 if mode == 'source' else out2
This part looks a little bit weird to me. Please correct me if I am wrong.
`out1` is the output from forwarding `data` to the source branch, and `out2` is the output from forwarding `data` to the target branch.
I thought we still use source branch for testing since the parameters learned in the source branch will be affected by the target branch through the CORAL loss. My concern is that **the target branch is not trained using the source labels**, so I am not sure whether it will produce the correct labels for the testing data. 
It would be great if you can clarify my concern, or you can show the result forwarding the testing data through the source branch.
Thank you!!!

**Comment by SSARCandy** [comment]
1. Averaging the loss have not special reason, just don't want to mess-up the log...

2. Actually, there is no difference between `out1` and `out2`, since the model parameters are shared between source and target, thanks to you for pointing it out.

those codes can be re-write as:

```
out, _ = model(data, data)
```

**Comment by cmhungsteve** [comment]
1. thank you for the explanation.
2. but the parameters of FC layer between the source and target branches are different, aren't they?

**Comment by SSARCandy** [comment]
![image](https://user-images.githubusercontent.com/6803791/35987920-087627a8-0d38-11e8-8971-5fb0b81c7615.png)
according to the original paper, they are all shared.

**Comment by cmhungsteve** [comment]
I see. I checked your `models.py` and found that although you have `source_fc` and `target_fc`, you only forward the data through `source_fc`. That's the same as what the paper does. I think I misunderstood in the first beginning. 
Thank you for your clarification!!

One more question.
Have you ever tried not to share the FC layer? I remember most of the other DA works don't share the parameters for FC layers.

**Comment by cmhungsteve** [comment]
By the way, how did you decide the batch size for source and target?
It's not mentioned in their paper.
Thanks

**Comment by SSARCandy** [comment]
my batch size is decided by tail and error, and the reason that source and target batch aren't same is due to the dataset lens (amazon are more than webccom).
And, yes I had tried independent `fc` layer, that's why my code has redundant `source_fc` and `target_fc` 😆 , the performance isn't good enough so I follow original paper setting eventually.
